components/workflow-mgt/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java

```diff
@@ -189,46 +181,50 @@ public void deleteHumanTask(WorkflowRequest workflowRequest) throws WorkflowImpl
-                        mutualSSLHeader = AXIOMUtil.stringToOM(headerString);
-                        client.addHeader(mutualSSLHeader);
-                    } catch (XMLStreamException e) {
-                        throw new AxisFault("Error while creating mutualSSLHeader XML Element.", e);
-                    }
+
+                    client = new HumanTaskClientAPIAdminClient(servicesUrl, bpsProfiles.get(i).getUsername());
                 } else {
-                    authenticate(client, bpsProfiles.get(i).getUsername(), bpsProfiles.get(i).getPassword());
+                    client = new HumanTaskClientAPIAdminClient(servicesUrl, bpsProfiles.get(i).getUsername(),
+                            bpsProfiles.get(i).getPassword());
                 }
-                TTaskSimpleQueryResultSet results = stub.simpleQuery(input);
+                TTaskSimpleQueryResultSet results = client.simpleQuery(input);
                 TTaskSimpleQueryResultRow[] arr = results.getRow();
-                for (int j = 0; j < arr.length; j++) {
-                    Object task = stub.getInput(arr[j].getId(), new NCName(""));
-                    InputStream stream = new ByteArrayInputStream(task.toString().getBytes(StandardCharsets.UTF_8));
-                    OMElement taskXML = new StAXOMBuilder(stream).getDocumentElement();
-                    Iterator<OMElementImpl> iterator = taskXML.getChildElements();
-                    while (iterator.hasNext()) {
-                        OMElementImpl child = iterator.next();
-                        checkMatchingTaskAndDelete(workflowRequest.getRequestId(), stub, arr, j, child);
-                    }
+                if (ArrayUtils.isNotEmpty(arr)) {
+                    for (int j = 0; j < arr.length; j++) {
+                        try {
+                            Object task = client.getInput(arr[j].getId());
+                            InputStream stream = new ByteArrayInputStream(task.toString().getBytes(StandardCharsets
+                                    .UTF_8));
+
+
+                            OMElement taskXML = new StAXOMBuilder(stream).getDocumentElement();
+                            Iterator<OMElementImpl> iterator = taskXML.getChildElements();
+                            while (iterator.hasNext()) {
+                                OMElementImpl child = iterator.next();
+                                checkMatchingTaskAndDelete(workflowRequest.getRequestId(), client, arr, j, child);
+                            }
+                        } catch (IllegalStateFault | XMLStreamException | IllegalArgumentFault | RemoteException |
+                                IllegalOperationFault | IllegalAccessFault e) {
+                            //If exception throws when retrieving and deleting a specific task, it will continue with
+                            // other tasks without terminating.
+                            log.info("Failed to check human task.");
```

Comment:
wouldn't it be a value addition to put extra information to identify what exactly failed. than a generic message 'Failed to check human task.'